Pseudocode/Dynamic Programming/Shortest subarray sorting which sorts entire array/SourceCode.tex

```diff
@@ -0,0 +1,80 @@
+% Set the Page Layout
+\documentclass[12pt]{article}
+\usepackage[inner = 2.0cm, outer = 2.0cm, top = 2.0cm, bottom = 2.0cm]{geometry}
+
+
+% Package to write pseudo-codes
+\usepackage{algorithm}
+
+% Remove the 'end' at the end of the algorithm
+\usepackage[noend]{algpseudocode}
+
+% Define Left Justified Comments
+\algnewcommand{\LeftComment}[1]{\Statex \(\triangleright\) #1}
+
+% Remove the Numbering of the Algorithm
+\usepackage{caption}
+\DeclareCaptionLabelFormat{algnonumber}{Algorithm}
+\captionsetup[algorithm]{labelformat = algnonumber}
+\newcommand{\Break}{\textbf{Break}}
+
+\begin{document}
+
+\begin{algorithm}
+
+  \caption{Function to find the shortest subarray sorting which sorts the entire array}
+  \begin{algorithmic}[1]
+    \Ensure One Based Indexing for the array
+    \Statex
+    \Function{$Find\_Sub\_Array$}{$arr$}
+        \LeftComment{$min\_right[i]$ denotes the maximum element to the right of i}
+        \LeftComment{$min\_left[i]$ denotes the maximum element to the left of i}
+        \Statex
+        
+        \State $n \gets arr.length$
+        \If{$n = 0$ \textbf{or} $n = 1$}
+            \State \Return{0}
+        \EndIf
+        
+        \Statex
+        \State $min\_right[n - 1] \gets arr[n - 1]$

```

Comment:
It looks like you've been confused by the notation. Remember, we are assuming one based indexing (See at the top of the function declaration). So, it should actually start from `n`